app/services/github_checks_verifier.rb

```diff
@@ -51,10 +54,10 @@ def fail_if_requested_check_never_run(check_name, all_checks)
     raise StandardError, "The requested check was never run against this ref, exiting..."
   end
 
-  def fail_unless_all_success(checks)
-    return if checks.all? { |check| check.conclusion == "success" }
+  def fail_unless_all_conclusions_allowed(checks)
+    return if checks.all? { |check| allowed_conclusions_array.include? check.conclusion }
 
-    raise StandardError, "One or more checks were not successful, exiting..."
+    raise StandardError, "One or more checks conclusions were not allowed, exiting..."
```

Comment:
Since this will be a message visible to the user, we could add some reference to the new input param, or add an entry to the documentation so it's easier to find. Something like: `"One or more checks conclude with unexpected statuses. The allowed/expected conclusion values for the checks are: [#{allowed_conclusions}]. This can be configured with the 'allowed_conclusions' param."`